rviz_common/include/rviz_common/message_filter_display.hpp

```diff
@@ -0,0 +1,202 @@
+ *
+ * This class handles subscribing and unsubscribing to a ROS node when the display is
+ * enabled or disabled. */
+template<class MessageType>
+class MessageFilterDisplay : public _RosTopicDisplay
+{
+// No Q_OBJECT macro here, moc does not support Q_OBJECT in a templated class.
+
+public:
+  /** @brief Convenience typedef so subclasses don't have to use
+   * the long templated class name to refer to their super class. */
+  typedef MessageFilterDisplay<MessageType> MFDClass;
+
+  MessageFilterDisplay()
+  : tf_filter_(nullptr),
+    messages_received_(0)
+  {
+    // TODO(Martin-Idel-SI): We need a way to extract the MessageType from the template to set a
+    // correct string. Previously was:
+    // QString message_type =
+    //   QString::fromStdString(message_filters::message_traits::datatype<MessageType>());
+    QString message_type = QString::fromStdString("");
+    topic_property_->setMessageType(message_type);
+    topic_property_->setDescription(message_type + " topic to subscribe to.");
+  }
+
+  /**
+  * When overriding this method, the onInitialize() method of this superclass has to be called.
+  * Otherwise, the ros node will not be initialized.
+  */
+  void onInitialize() override
+  {
+    _RosTopicDisplay::onInitialize();
+  }
+
+  ~MessageFilterDisplay() override
+  {
+    unsubscribe();
+  }
+
+  void reset() override
+  {
+    Display::reset();
+    if (tf_filter_) {
+      tf_filter_->clear();
+    }
+    messages_received_ = 0;
+  }
+
+  void setTopic(const QString & topic, const QString & datatype) override
+  {
+    (void) datatype;
+    topic_property_->setString(topic);
+  }
+
+protected:
+  void updateTopic() override
+  {
+    unsubscribe();
+    reset();
+    subscribe();
+    context_->queueRender();
+  }
+
+  virtual void subscribe()
+  {
+    if (!isEnabled()) {
+      return;
+    }
+
+    if (topic_property_->isEmpty()) {
+      setStatus(properties::StatusProperty::Error,
+        "Topic",
+        QString("Error subscribing: Empty topic name"));
+      return;
+    }
+
+    try {
+      subscription_ = std::make_shared<message_filters::Subscriber<MessageType>>(
+        rviz_ros_node_.lock()->get_raw_node(), topic_property_->getTopicStd(), qos_profile);
+      tf_filter_ =
+        std::make_shared<tf2_ros::MessageFilter<MessageType, transformation::FrameTransformer>>(
+        *context_->getFrameManager()->getTransformer(),
+        fixed_frame_.toStdString(), 10, rviz_ros_node_.lock()->get_raw_node());
+      tf_filter_->connectInput(*subscription_);
+      tf_filter_->registerCallback(std::bind(&MessageFilterDisplay<MessageType>::incomingMessage,
+        this, std::placeholders::_1));
+      setStatus(properties::StatusProperty::Ok, "Topic", "OK");
+    } catch (rclcpp::exceptions::InvalidTopicNameError & e) {
+      setStatus(properties::StatusProperty::Error, "Topic",
+        QString("Error subscribing: ") + e.what());
+    }
+  }
+
+  void transformerChangedCallback() override
```

Comment:
The implementation of this method is identical to `updateTopic`. Maybe it's worth consolidating the logic in a common `resetSubscription()` method?

Reply:
It looks like that `override` makes the compilation fail.
```
In file included from /home/lizo/ros_packaging_linux/ros2_overlay/src/rviz/rviz_default_plugins/include/rviz_default_plugins/displays/pointcloud/point_cloud_scalar_display.hpp:42:0,
                 from /home/lizo/ros_packaging_linux/ros2_overlay/src/rviz/rviz_default_plugins/include/rviz_default_plugins/displays/illuminance/illuminance_display.hpp:34,
                 from /home/lizo/ros_packaging_linux/ros2_overlay/src/rviz/rviz_default_plugins/src/rviz_default_plugins/displays/illuminance/illuminance_display.cpp:31:
/home/lizo/ros_packaging_linux/ros2_overlay/install/rviz_common/include/rviz_common/message_filter_display.hpp: In instantiation of ‘class rviz_common::MessageFilterDisplay<sensor_msgs::msg::Illuminance_<std::allocator<void> > >’:
/home/lizo/ros_packaging_linux/ros2_overlay/src/rviz/rviz_default_plugins/include/rviz_default_plugins/displays/pointcloud/point_cloud_scalar_display.hpp:65:35:   required from ‘class rviz_default_plugins::displays::PointCloudScalarDisplay<sensor_msgs::msg::Illuminance_<std::allocator<void> > >’
/home/lizo/ros_packaging_linux/ros2_overlay/src/rviz/rviz_default_plugins/include/rviz_default_plugins/displays/illuminance/illuminance_display.hpp:51:12:   required from here
/home/lizo/ros_packaging_linux/ros2_overlay/install/rviz_common/include/rviz_common/message_filter_display.hpp:134:8: error: ‘void rviz_common::MessageFilterDisplay<MessageType>::transformerChangedCallback() [with MessageType = sensor_msgs::msg::Illuminance_<std::allocator<void> >]’ marked ‘override’, but does not override
   void transformerChangedCallback() override
```

Reply:
Wonder how does it build without any errors for me

Reply:
The `override` keyword makes sense to me, and it also builds for me. Perhaps you were building `rviz_default_plugins` against an older version of `rviz_common` somehow?